### ISSUE TITLE
Error building from source code.

### ISSUE THREAD

**Issue by Osmanyasal** [performancecopilot/pcp#1796]
Hello, 
I'm trying to install PCP on my Centos 7 system, since the default repo installs 4.3-ish (old version) of PCP and I need the newest version of it, I'm trying to install it by building from source code however I get the following error.

I follow the instructions here : https://github.com/performancecopilot/pcp/blob/main/INSTALL.md
I get the error after running ``` ./Makepkgs --verbose ``` command 

```
Finding  Requires: /home/oyasal22/pcp/pcp-6.1.0/pcp-6.1.0/build/rpm/filter-requires -v redhat -f 0 -r 07 /usr/lib/rpm/redhat/find-requires
Provides: pcp-export-zabbix-agent = 6.1.0-1 pcp-export-zabbix-agent(x86-64) = 6.1.0-1 zbxpcp.so()(64bit) zbxpcp.so(ZBXPCP_1.0)(64bit)
Requires(rpmlib): rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1 rpmlib(CompressedFileNames) <= 3.0.4-1
Requires: libc.so.6()(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libdl.so.2()(64bit) libdl.so.2(GLIBC_2.2.5)(64bit) libpcp.so.3()(64bit) libpcp.so.3(PCP_3.0)(64bit) libpcp.so.3(PCP_3.13)(64bit)
Processing files: pcp-export-pcp2elasticsearch-6.1.0-1.x86_64
error: Could not open %files file /home/oyasal22/pcp/pcp-6.1.0/pcp-6.1.0/pcp-export-pcp2elasticsearch-files.rpm: No such file or directory


RPM build errors:
    File listed twice: /usr/share/doc/pcp-6.1.0
    File listed twice: /usr/share/doc/pcp-6.1.0/CHANGELOG
    File listed twice: /usr/share/doc/pcp-6.1.0/COPYING
    File listed twice: /usr/share/doc/pcp-6.1.0/INSTALL.md
    File listed twice: /usr/share/doc/pcp-6.1.0/README.md
    File listed twice: /usr/share/doc/pcp-6.1.0/VERSION.pcp
    File listed twice: /usr/share/doc/pcp-6.1.0/pcp.lsm
    Could not open %files file /home/oyasal22/pcp/pcp-6.1.0/pcp-6.1.0/pcp-export-pcp2elasticsearch-files.rpm: No such file or directory
make: *** [pack_pcp] Error 1
make: Leaving directory `/home/oyasal22/pcp/pcp-6.1.0/build/rpm'

```

**Comment by kmcdonell** [comment]
@Osmanyasal Not sure what's happened here, as we're building PCP regularly on all versions of CentOS back to 6.10.

First guess would be that some optional part of the build is not quite meshing with the packages you have installed on the build machine.

Could you please add the following (both run from the top-level of the PCP source tree after Makepkgs has been run):
1. output from $ grep -i 'python.*requests' Logs/pcp
2. output from $ qa/admin/list-packages -m -i cpan

**Comment by Osmanyasal** [comment]
[oyasal22@du01 pcp]$ grep -i 'python.*requests' Logs/pcp
checking for Python module requests... yes
checking for Python module requests... no

[oyasal22@du01 pcp]$ sudo qa/admin/list-packages -m -i cpan
cpan(Spreadsheet::Read) 

for second command I'd to cancel after 5mins with no result, when i run with sudo it gave the output above.

https://pcp.readthedocs.io/en/latest/HowTos/installation/index.html
I tried updating the dnf path as described here but it still tries to install 4.5, if there's a config for 6.1 I can take that as well.

**Comment by kmcdonell** [comment]
Thanks @Osmanyasal.

The only reason for the delay in list-packages that I can think of is that "rpm -qa" is triggering a refresh of the available packages from the repos ... this might explain why it worked with sudo.  But I made a mistake in the command I asked you to run, it should have been -x cpan not -i cpan ... are you able to try that part again?

More seriously the Python module request lines show a problem.  We run configure twice in the build and SHOULD get the same answers in both cases.  I don't understand how this can be different, nor why even with the answer "no" in the rpm-rebuild the build would fail.

I've just tried again on my CentOS 7 system, and the configure answer is "yes" twice.  And the pcp-export-pcp2elasticsearch rpm gets built without an issue.

Are you able to upload the file pcp-6.1.0/src/include/builddefs after the failed build and attach it to this issue?  That (in addition to the amended list-packages -m output) would help me focus on the differences between your build and mine.

**Comment by Osmanyasal** [comment]
sure, the output of the command is as follows
`[oyasal22@du01 pcp]$ sudo qa/admin/list-packages -m -x cpan
[sudo] password for oyasal22: 
389-ds 389-ds-base Coin3-devel avahi-devel avahi-tools bcc-tools cppcheck cyrus-sasl-devel device-mapper-devel docker docker-common gfs2-utils httpd java-latest-openjdk-headless kubernetes-client kubernetes-node libcmocka-devel libicu-devel libuv-devel libvirt-python lm_sensors mandoc memcached nmap openssl11 openssl11-devel perl-Date-Manip perl-File-Slurp perl-JSON perl-LDAP perl-List-MoreUtils perl-Net-SNMP perl-Spreadsheet-WriteExcel perl-Spreadsheet-XLSX perl-Text-CSV_XS perl-XML-LibXML perl-XML-TokeParser perl-YAML-LibYAML perl-libwww-perl postfix-perl-scripts postgresql pylint python3-psycopg2 python36-lxml python36-pillow python36-requests python36-six qt5-qtbase-devel qt5-qtsvg-devel ragel redis rrdtool rrdtool-perl selinux-policy-devel setools-console socat zabbix22-agent 
`
and the aforementioned file is as follows
[builddefs.txt](https://github.com/performancecopilot/pcp/files/12386935/builddefs.txt)

**Comment by kmcdonell** [comment]
OK @Osmanyasal that's helped, but I still don't understand your build failure.

Any chance you can upload Logs/pcp from the build ... it is 3+ MB?  We may need help from Google Drive or Dropbox if that's too big for an attachment to github issue.

Aside from that, do you have both python2 and python3 installed?

To build the *same* as we are, you'd need to install all of the packages reported by list-packages -m ... it *should* build OK without these ones (although it demonstrably does not and I'd like to understand why), but without this long list of packages installed on the build machine you won't get all of the optional PCP rpms built and/or won't be able to run all of the QA tests.

And a wild-arsed-guess for a simpler approach would be to try installing just the python36-requests package and trying again.

Thanks for sticking with this.

**Comment by kmcdonell** [comment]
Hmm @Osmanyasal.
I've removed all the packages from your list-packages -m list and my build completes without error (no pcp2elastic rpm, but that's expected in this scenario).
So it must be something *else* that is different on your build system.

**Comment by Osmanyasal** [comment]
Thank you, sir. After installing the python36-request package, I could successfully install PCP. 

![image](https://github.com/performancecopilot/pcp/assets/22853419/7bc92635-0897-4c79-a738-1400591e86de)

**Comment by kmcdonell** [comment]
@Osmanyasal Do you still have Logs/pcp from the failed build?
I'd like to triage this some more, but if we don't have that log file and I cannot reproduce the build failure, the only real option will be to close the issue.

**Comment by Osmanyasal** [comment]
yes I've 
this is the log file BEFORE I installed the aforementioned package. If you wish, I can send you the one related to the successful build.

[pcp.txt](https://github.com/performancecopilot/pcp/files/12387997/pcp.txt)

**Comment by Osmanyasal** [comment]
hello sir, there's another issue with profiling data with pcp2influxdb
I have the pmcd service up and running, I can successfully ping the machine as well, yet when I try to monitor some metrics with a pcp2influxdb configuration file I get the following error.
``` 
pcp2influxdb -t 1 -c pcp_lab_intel_monitor.conf -5 :configured
pcp2influxdb: Connection refused; is pmcd running?
```
![Screenshot from 2023-08-20 17-57-13](https://github.com/performancecopilot/pcp/assets/22853419/e96a9b55-4501-4e7a-ac48-8bd886bdc3cf)



I'm adding the config file here 
[pcp_intel_lab_monitor.con.txt](https://github.com/performancecopilot/pcp/files/12388457/pcp_intel_lab_monitor.con.txt)
we generate these files automatically and since I've not installed pcp from its sourcecode, I belive this is a related issue to this.

**Comment by kmcdonell** [comment]
@Osmanyasal what is the IP addr of du01.ku.edu.tr?

**Comment by Osmanyasal** [comment]
inside the file I shared, it's 172.20.31.81 however it's a closed network. I use my school's VPN to connect.
I personally don't expect the config file to be wrong since we generate it automatically per computer, so far so good, however, I don't think it's PCP either since the service is up and running.

**Comment by kmcdonell** [comment]
Is your VPN allowing connections on pmcd's default port (44321)?
On the machine where you're running pcp2influxdb, try this:
```
pminfo -h 172.20.31.81 -f -Dcontext,pmapi hinv.ncpu
```
and add the output to this issue.

**Comment by kmcdonell** [comment]
@Osmanyasal OK, next let's look at /etc/pcp/pmcd/pmcd.conf ... in particular the [access] section.  If you have something like
```
allow "local:*" : all;
```
you'll need to replace it with
```
allow ".*" : all;
```
for **any** IPv4 host to be able to access pmcd, else maybe
```
allow your-ipv4-addr:all;
```
if you want to only allow access from your-ipv4-addr.

**Comment by Osmanyasal** [comment]
I still get the same error, 
I replaced as you said however when I restarted the service ``` sudo systemctl restart pmcd ``` it gave an error then I removed dot (assumed it's a typo) and typed ```allow "*" : all;``` I restarted however It still give the same error

![image](https://github.com/performancecopilot/pcp/assets/22853419/fd89bf1c-a6be-411a-8b58-18337b82914f)
![image](https://github.com/performancecopilot/pcp/assets/22853419/b7a6080b-410e-4e17-a5b5-9b649ea9fe27)
![image](https://github.com/performancecopilot/pcp/assets/22853419/3e3edf89-38b0-49f7-ae24-e3061f732fde)
![image](https://github.com/performancecopilot/pcp/assets/22853419/06b1c26b-ebbd-4869-bc14-397aa2f24b40)

this is my own PC that i monitor the target system
![image](https://github.com/performancecopilot/pcp/assets/22853419/86883e69-fbb6-4dc0-8137-820c03e2c0ac)

**Comment by kmcdonell** [comment]
@Osmanyasal Sorry, but this looks most like a firewall issue.
From your laptop, can you please try:
```
$ telnet -4 172.20.31.81 44321
```
if it is **not** a firewall issue you should see something like this:
```
kenj@bozo-vm:~/src/pcp$ telnet -4 bozo 44321
Trying 192.168.178.100...
Connected to bozo.localdomain.
Escape character is '^]'.
pU
```
It is the "Connected" part that's important ... the "pU" part is the (binary) handshake between pmcd and a client and may be different.

If this does not shed any light, then the last thing I can suggest is to turn on some instrumentation for pmcd ...

1. edit /etc/pcp/pmcd/pmcd.options and add a line with -Dauth,appl3
2. restart pmcd
3. try your client connection from the laptop
4. inspect the /var/log/pcp/pmcd/pmcd.log file on the system where pmcd is running, and attach it to this issue.

**Comment by kmcdonell** [comment]
@Osmanyasal Has this issue been resolved?

And just a note on the [access] clause, ".\*" is **not** a typo, it matches any IPv4 address (this is not a regexp(3) nor a shell glob expression, it is described in pmcd(1)).  "\*" is also syntactically correct, but it will match *any* IPv4 *or* IPv6 address.

**Comment by Osmanyasal** [comment]
Hello,

Yes It has been resolved

**Comment by kmcdonell** [comment]
OK, nothing more to see here.
